src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -2455,6 +2456,14 @@ Socket findSocket(String hostName, int portNumber, int timeoutInMilliSeconds, bo
         try {
             InetAddress[] inetAddrs = null;
 
+            // Used in testing to force multiple InetAddress and thus findSocketUsingThreading.
+            // Cleanest way to implement this is for this test condition to be the first one checked.
+            if (hostName.equals("testServerName")) {
```

Comment:
Hmm, maybe we should look into using flags to gate code specifically for testing. Maybe something like a JVM arg `-DtestMode`.

Reply:
Do we already do this anywhere?

Reply:
Agreed. This doesn't feel right to me...

Reply:
Ok, I'll take a look into this.

Reply:
don't put test code into the test driver, do this in the tests

Reply:
I agree, I'll revisit prior methods of testing to see if there's something I missed.